components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+/* global $ */
+define([
+    './helpers/create-element.js',
+    './helpers/cancel-event.js',
+    './helpers/clear-element.js',
+    './helpers/component-support.js',
+    './helpers/post-via-iframe.js',
+    './helpers/trace.js',
+    './helpers/validate-against-schema.js',
+    './helpers/to-camel-case.js',
+    'document-register-element'
+], (createElement, cancelEvent, clearElement, componentSupport, postViaIframe, trace, validateAgainstSchema, toCamelCase) => {
+
+    //
+    // TODO: consider adding method to validate attachment data against headers before firing external events
+    //
+    /**
+     * Class Import for a ca-import web component
+     * @extends HTMLElement
+     */
+    class Import extends HTMLElement {
+        /**
+         * Executes when the element is first created
+         * @returns {undefined} nothing
+         */
+        createdCallback() {
+            const label = createElement(this, 'label', {});
+            createElement(label, 'span', {});
+            this.link = createElement(label, 'a', { class: 'ca-import-download', href: '#', target: '_target', download: 'template.csv' }, 'Download Template');
+            this.form = createElement(label, 'form');
+            createElement(this.form, 'div', {}, 'Upload');
+            this.input = createElement(this.form, 'input', { name: 'ca-import-upload', type: 'file', accept: '.csv' });
+            const table = createElement(this, 'table', {});
+            createElement(table, 'colgroup', {});
+            createElement(table, 'thead', {});
+            createElement(table, 'tbody', {});
+
+            this.input.onchange = this.fileHandler.bind(this);
+            this.input.onclick = this.fileClick.bind(this);
+            this.link.onclick = this.downloadTemplateHandler.bind(this);
+        }
+
+        /**
+         * Executes when any attribute is changed on the element
+         * @param  {string} attrName - the name of the attribute to have changed
+         * @param  {string} oldVal - the old value of the attribute
+         * @param  {string} newVal - the new value of the attribute
+         * @returns {void} void
```

Comment:
That was a copy pasta from the original since it didn't throw a lint error. I prefer undefined since that's the return type.